tests/python/unittest/test_target_codegen_hexagon.py

```diff
@@ -26,8 +26,10 @@
 import tvm.contrib.hexagon as hexagon
 
 
-# Register a phony linker, so that we can test codegen without a Hexagon toolchain.
-hexagon.register_linker(lambda: "/bin/true")
+@pytest.fixture(scope="session", autouse=True)
+def register_linker():
+    # Register a phony linker, so that we can test codegen without a Hexagon toolchain.
+    hexagon.register_linker(lambda: "/bin/true")
```

Comment:
don't you need to unregister it after the test? e.g. https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended and then un-register after the yield

Reply:
This changes the global state of the TVM runtime (the registry), so I'm not sure how this is going to work if there are other tests running at the same time (that would want to set it to something else).  I'm not really anticipating adding runtime tests for Hexagon (only codegen) in the foreseeable future, so this "permanent" registration should work.

The only reason the linker is needed is to create a shared library in `BuildHexagon`.  We can do all the preceding steps (incl. code generation into an object file) without any part of the Hexagon toolchain.  I think a better solution would be to see if we can delay the creation of the shared library until it needs to be saved.  This would eliminate the need for registering the linker altogether in codegen tests.

Reply:
tests don't run concurrently in the same python instance. if we do run tests concurrently, we use a different python interpreter (e.g. with xdist). a lot more stuff is gonna break if we run two tests concurrently in tvm. 

that said, i agree a better solution is probably to pass in the linker path rather than relying on system linker. up to you which route you want to go here--passing in the linker path is likely more invasive.

Reply:
@areusch I believe there is a request here to unregister the global override? @kparzysz-quic let us know if that's something that can be modified in this PR.

Reply:
yes, it would be great if we can set the global func back the way it was before this test ran. i would like to avoid any leftover state after we switch to xdist as the exact ordering between tests will become less clear PR-to-PR.

Reply:
@kparzysz-quic looks like this is a requested change: let us know if you have any questions